nautobot/project-static/js/forms.js

```diff
@@ -672,6 +696,39 @@ function initializeDynamicFilterForm(context){
         $("#dynamic-filter-form").submit()
     })
 
+    // On clear of filter form
+    this_context.find("#dynamic-filter-form, #default-filter form").on("reset", function(e){
+        console.log("hello")
+        e.preventDefault()
+        const urlParams = new URLSearchParams(window.location.search);
+        const saved_view_pk = urlParams.get('saved_view');
+        var saved_view_pk_query_string = ""
+        if (saved_view_pk != null){
+            saved_view_pk_query_string = "&saved_view=" + saved_view_pk
+        }
+        const sort_order = urlParams.getAll('sort');
+        var sort_order_query_string = ""
+        if (sort_order != null){
+            for (var i = 0; i < sort_order.length; i++){
+                appendage = "&sort=" + sort_order[i]
+                sort_order_query_string += appendage
+            }
+        }
+        const per_page = urlParams.get('per_page');
+        var per_page_query_string = ""
+        if (per_page_query_string != null) {
+            per_page_query_string = "per_page=" + per_page
+        }
+        const table_changes = urlParams.get('table_changes_pending');
+        var table_changes_query_string = ""
+        if (table_changes_query_string != null) {
+            table_changes_query_string = "&table_changes_pending=" + table_changes
+        }
+        // clear filter params info and retain other information
+        location.replace("?" + per_page_query_string + sort_order_query_string + saved_view_pk_query_string + table_changes_query_string)
```

Comment:
Let's use `location.assign()` instead of `location.replace()` so that back button works as expected.